SofaKernel/modules/SofaBaseMechanics/SubsetMapping.inl

```diff
@@ -187,7 +191,15 @@ template <class TIn, class TOut>
 void SubsetMapping<TIn, TOut>::apply ( const core::MechanicalParams* /*mparams*/, OutDataVecCoord& dOut, const InDataVecCoord& dIn )
 {
     const IndexArray& indices = f_indices.getValue();
-
+    
+    if (f_resizeToModel.getValue() || this->toModel->getSize() < indices.size())
+    { 
+        if (this->toModel->getSize() != indices.size()) 
```

Comment:
You are comparing a != after a < (always true).

Personally, I would do like any other mappings, i.e. removing the f_resizeToModel Data (that should be named with a d_ BTW) and simply perform `this->toModel->resize(indices.size()); ` all the time. It would be way more clear, and possibly more efficient than evaluating Data and the user would not have to activate anything special.